lib/grp.gi

```diff
@@ -2685,6 +2685,9 @@ end );
 InstallMethod( NaturalHomomorphism, "for a group with natural homomorphism stored",
     [ IsGroup ],
 function(G)
+  Error("The use of `NaturalHomomorphism` for a `FactorGroup`\n",
+    "has been removed, as it caused side-effects.");
```

Comment:
But won't that break existing user code that relies on this?

Reply:
Yes, should code rely onthis functionality. We could change it to add text "enter `return;` to continue, or change the error to a warning message.

Reply:
Also we should perhaps quickly check if any distributed packages rely on this?

Reply:
I have checked through the packages in 4.11. Only XMod,  laguna, primgrp, rds, wedderga and xgap call `FactorGroup(`. None of them uses `NaturalHomomorphism(`. I will change the error to a warning.

Reply:
XMod does use NaturalHomomorphism

Reply:
@cdwensley Where does it use `NaturalHomomorphism`? I found it uses `NaturalHomomorphismByNormalSubgroup` (which is safe and well) only in version 2.82.

Reply:
I'm happy to change it if that is the sensible thing to do. 
Christopher-Wensleys-iMac:lib christopherwensley$ grep -n NaturalHomomorphism *
catone.gi:575:                nat:=NaturalHomomorphismByNormalSubgroup(G,N);
gp2map.gi:1764:    nat1 := NaturalHomomorphism( KR1 ); 
gp2map.gi:1795:                          nat2 := NaturalHomomorphism( KQ ); 
gp2map.gi:1862:    nat1 := NaturalHomomorphism( KR1 ); 
gp2map.gi:1891:                        natKQ := NaturalHomomorphism( KQ );
gp2map.gi:1897:                        natKR1 := NaturalHomomorphism( KR1 );
gp2map.gi:1902:                        natKQ := NaturalHomomorphism( KQ );

Reply:
Yes, this does appear to be a recent change.

Reply:
Since this is a recent change (and thus possibly not even in a distributed version) I would urge to replace code of the form
```
F:=FactorGroup(G,N);
hom:=NaturalHomomorphism(F);
```
by
```
hom:=NaturalHomomorphismByNormalSubgroup(G,N);
F:=Image(hom,G);
```
The cost will be the same, but it does not rely on the side-effect of aching the homomorphism in the object.

Reply:
It all happened when quasi-isomorphisms were introduced: PR #107 (17/2/21). 
I'll try using NaturalHomomorphismByNormalSubgroup instead.